zerver/models.py

```diff
@@ -550,6 +555,16 @@ class PreregistrationUser(models.Model):
 
     realm = models.ForeignKey(Realm, null=True) # type: Optional[Realm]
 
+class OutgoingWebhookBot(UserProfile):
+    base_url = models.TextField() # type: text_type
+    service_api_key = models.TextField() # type: text_type
```

Comment:
Hmm, for generic outgoing webhooks services, the variable API key is generated by the service making the outgoing requests, not the service receiving them; for example in Slack's version, https://api.slack.com/outgoing-webhooks, the token is by default provided by Slack, not the third-party service.  

However, I think maybe what you had in mind is the case where we're doing an outgoing webhook integration for the specific third-party service like some of the bot examples, where the third-party service might have a token format it wants, right?  OK, that makes sense, I'm on board with sticking with TextField here.  Thanks for the explanation!

Given that, I guess I'm on board with making this a Text Field.